### ISSUE TITLE
Add support for ```Intl.ListFormat```

### ISSUE THREAD

**Issue by samhenrigold** [oven-sh/bun#808]
### Version

0.1.4

### Platform

Darwin mbp.local 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000 arm64

### What steps will reproduce the bug?

```
// plural.js

let fullNames = ['Jack Sims', 'Nicole Kim', 'Carl Shaw', 'Michael Hayes', 'Lori Holmes', 'Denise Robinson'];
const formatter = new Intl.ListFormat('en-US', { style: 'long', type: 'conjunction' });
console.log(formatter.format(fullNames));
```

### What is the expected behavior?

This is the correct output from Node:
```
$ node plural.js                      
Jack Sims, Nicole Kim, Carl Shaw, Michael Hayes, Lori Holmes, and Denise Robinson
```
WebKit has the same output when tested in the browser console.

### What do you see instead?

```
~/Desktop $ bun run plural.js

1 | let fullNames = ['Jack Sims', 'Nicole Kim', 'Carl Shaw', 'Michael Hayes', 'Lori Holmes', 'Denise Robinson'];
2 | const formatter = new Intl.ListFormat('en-US', { style: 'long', type: 'conjunction' });
                      ^
 TypeError: undefined is not a constructor (evaluating 'new Intl.ListFormat("en-US", { style: "long", type: "conjunction" })')
      at /Users/sam/Desktop/plural.js:2:18
```

**Comment by PeterEckIII** [comment]
Is there an update on this issue? I'd like to begin using Bun with Remix. Thanks

**Comment by birkskyum** [comment]
@PeterEckIII  here https://github.com/oven-sh/bun/issues/1843#issuecomment-1710732503

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v1.0.3, thanks to @dylan-conway for upgrading WebKit and to @constellation for upgrading ICU in WebKit 

<img width="271" alt="image" src="https://github.com/oven-sh/bun/assets/709451/81a2334b-cde8-4870-99d1-968c3123916d">
